Remove proc-macro-error as an abandonware dependency

I ran into an error in my CI/CD testing. I am performing a test where I compile all crate dependencies using the minimum valid version numbers. The bilge-impl crate fails to compile - although the error is in its dependency, the proc-macro-error crate. The proc-macro-error crate has a bug in version 1.0.0 which is fixed in its latest version, 1.0.4. However, because bilge-impl specifies the minimum proc-macro-error version as 1.0 instead of ^1.0.4, this error shows up when I do the minimum-version test.

The quick and easy fix is to specify the version number, and just change the bilge-impl Cargo.toml file dependency to `proc-macro-error = { version = "1.0.4", default-features = false }`

I think that the quick fix would be wrong, however - it's clear that the proc-macro-error crate is abandoned. It doesn't have any updates since 3 years ago, its owner has no git history over the last year, and it is failing to keep up with some critical libraries - syn 2.0, for example.

To finish this out, I think that 
(1) any proc-macro-error usage should be taken out of the codebase.
(2) this library should also implement a similar minimum version CI/CD test.

I'd be happy to help with both and submit a PR.

My compile error, shown in my [CI/CD pipeline](https://github.com/johngigantic/allegro-motor-drivers/actions/runs/6527472698/job/17722371098)

```
cargo check --all-features --locked
...
   Compiling bilge-impl v0.2.0
error: expected one of `.`, `;`, `?`, `else`, or an operator, found `::`
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bilge-impl-0.2.0/src/try_from_bits.rs:35:9
   |
35 | ...   emit_call_site_warning!("enum fills its bitsize"; help = "you can use `#[derive(FromBits)]` instead, rust will provide `TryFrom` for you (so you don't necessarily have to update call-sites)...
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected one of `.`, `;`, `?`, `else`, or an operator
   |
   = note: this error originates in the macro `emit_call_site_warning` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `bilge-impl` (lib) due to previous error
Error: Process completed with exit code 101.
```

--- a/bilge-impl/Cargo.toml
+++ b/bilge-impl/Cargo.toml
@@ -23,7 +23,7 @@ serde = []
 syn = { version = "2.0", features = ["full"] }
 quote = "1.0"
 proc-macro2 = "1.0"
-proc-macro-error = { version = "1.0", default-features = false }
+proc-macro-error2 = { version = "2.0", default-features = false }
 itertools = ">=0.11.0, <=0.14"
 
 [dev-dependencies]

--- a/bilge-impl/src/bitsize.rs
+++ b/bilge-impl/src/bitsize.rs
@@ -1,7 +1,7 @@
 mod split;
 
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::{abort, abort_call_site};
+use proc_macro_error2::{abort, abort_call_site};
 use quote::quote;
 use split::SplitAttributes;
 use syn::{punctuated::Iter, spanned::Spanned, Fields, Item, ItemEnum, ItemStruct, Type, Variant};

--- a/bilge-impl/src/bitsize/split.rs
+++ b/bilge-impl/src/bitsize/split.rs
@@ -1,4 +1,4 @@
-use proc_macro_error::{abort, abort_call_site};
+use proc_macro_error2::{abort, abort_call_site};
 use quote::ToTokens;
 use syn::{meta::ParseNestedMeta, parse_quote, Attribute, Item, Meta, Path};
 

--- a/bilge-impl/src/debug_bits.rs
+++ b/bilge-impl/src/debug_bits.rs
@@ -1,5 +1,5 @@
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::abort_call_site;
+use proc_macro_error2::abort_call_site;
 use quote::quote;
 use syn::{Data, Fields};
 

--- a/bilge-impl/src/default_bits.rs
+++ b/bilge-impl/src/default_bits.rs
@@ -1,5 +1,5 @@
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::abort_call_site;
+use proc_macro_error2::abort_call_site;
 use quote::quote;
 use syn::{Data, DeriveInput, Fields, Type};
 

--- a/bilge-impl/src/from_bits.rs
+++ b/bilge-impl/src/from_bits.rs
@@ -1,6 +1,6 @@
 use itertools::Itertools;
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::{abort, abort_call_site};
+use proc_macro_error2::{abort, abort_call_site};
 use quote::quote;
 use syn::{punctuated::Iter, Data, DeriveInput, Fields, Type, Variant};
 

--- a/bilge-impl/src/lib.rs
+++ b/bilge-impl/src/lib.rs
@@ -1,5 +1,5 @@
 use proc_macro::TokenStream;
-use proc_macro_error::proc_macro_error;
+use proc_macro_error2::proc_macro_error;
 
 mod bitsize;
 mod bitsize_internal;

--- a/bilge-impl/src/serde_bits.rs
+++ b/bilge-impl/src/serde_bits.rs
@@ -1,6 +1,6 @@
 use itertools::MultiUnzip;
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::abort_call_site;
+use proc_macro_error2::abort_call_site;
 use quote::quote;
 use syn::{Data, Field, Fields};
 

--- a/bilge-impl/src/shared.rs
+++ b/bilge-impl/src/shared.rs
@@ -4,7 +4,7 @@ pub mod util;
 
 use fallback::{fallback_variant, Fallback};
 use proc_macro2::{Ident, Literal, TokenStream};
-use proc_macro_error::{abort, abort_call_site};
+use proc_macro_error2::{abort, abort_call_site};
 use quote::quote;
 use syn::{Attribute, DeriveInput, LitInt, Meta, Type};
 use util::PathExt;

--- a/bilge-impl/src/shared/discriminant_assigner.rs
+++ b/bilge-impl/src/shared/discriminant_assigner.rs
@@ -1,5 +1,5 @@
 use proc_macro2::Literal;
-use proc_macro_error::abort;
+use proc_macro_error2::abort;
 use syn::{Expr, ExprLit, Lit, Variant};
 
 use super::{unreachable, BitSize};

--- a/bilge-impl/src/shared/fallback.rs
+++ b/bilge-impl/src/shared/fallback.rs
@@ -1,6 +1,6 @@
 use itertools::Itertools;
 use proc_macro2::Ident;
-use proc_macro_error::{abort, abort_call_site};
+use proc_macro_error2::{abort, abort_call_site};
 use syn::{Data, Variant};
 
 use super::{bitsize_from_type_ident, is_fallback_attribute, last_ident_of_path, unreachable, BitSize};

--- a/bilge-impl/src/try_from_bits.rs
+++ b/bilge-impl/src/try_from_bits.rs
@@ -1,5 +1,5 @@
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::{abort, emit_call_site_warning};
+use proc_macro_error2::{abort, emit_call_site_warning};
 use quote::quote;
 use syn::{punctuated::Iter, Data, DeriveInput, Fields, Type, Variant};
 

